src/filtrering/lagrede-filter-controller.tsx

```diff
@@ -0,0 +1,60 @@
+import {useEffect} from "react";
+import {
+    avmarkerSisteValgtMineFilter,
+    avmarkerValgtMineFilter,
+    avmarkerValgtVeilederGruppe,
+    markerMineFilter,
+    markerValgtVeilederGruppe
+} from "../ducks/lagret-filter-ui-state";
+import {erObjektValuesTomt, lagretFilterValgModellErLik} from "../components/modal/mine-filter/mine-filter-utils";
+import {useDispatch, useSelector} from "react-redux";
+import {AppState} from "../reducer";
+import {logEvent} from "../utils/frontend-logger";
+import {finnSideNavn} from "../middleware/metrics-middleware";
+import {ListevisningType} from "../ducks/ui/listevisning";
+import {veilederlisterErLik} from "../components/modal/veiledergruppe/veileder-gruppe-utils";
+
+export function LagredeFilterUIController(props: { filtergruppe: ListevisningType }) {
+    const dispatch = useDispatch()
+
+    const filtreringEnhetensOversikt = useSelector((state: AppState) => state.filtreringEnhetensOversikt);
+    const filtreringMinoversikt = useSelector((state: AppState) => state.filtreringMinoversikt);
+    const filtreringVeilederoversikt = useSelector((state: AppState) => state.filtreringVeilederoversikt);
+
+    const lagretMineFilter = useSelector((state: AppState) => state.mineFilter.data);
+    const lagretVeilederGrupper = useSelector((state: AppState) => state.veiledergrupper.data);
+
+    useEffect(() => {
+        const getFiltrering = () => {
+            if (props.filtergruppe === ListevisningType.veilederOversikt) return filtreringVeilederoversikt;
+            else if (props.filtergruppe === ListevisningType.minOversikt) return filtreringMinoversikt;
+            else if (props.filtergruppe === ListevisningType.enhetensOversikt) return filtreringEnhetensOversikt;
+        }
+
+        const valgtMineFilter = lagretMineFilter.filter(elem => lagretFilterValgModellErLik(elem.filterValg,getFiltrering()))
+        const valgtVeilederGruppe = lagretVeilederGrupper.filter(elem => veilederlisterErLik(elem.filterValg.veiledere,getFiltrering()!.veiledere))
```

Comment:
is `veilederlisteErLik` the same function as `lagretFilterValgModellErLik`?

Reply:
no no, since _veilederlisteErLik_ is just comparing _veiledere_ list inside filter valg, and _lagretFilterValgModellErLik_ is comparing everything else.

Reply:
Ahaa, I understand, thanks!